codec-http/src/main/java/io/netty/handler/codec/http/multipart/DefaultHttpDataFactory.java

```diff
@@ -241,7 +248,17 @@ public FileUpload createFileUpload(HttpRequest request, String name, String file
     public void removeHttpDataFromClean(HttpRequest request, InterfaceHttpData data) {
         if (data instanceof HttpData) {
             List<HttpData> fileToDelete = getList(request);
```

Comment:
this method's usage of `getList` seems like it maybe problematic. `getList` will create a new list (which contains `request`) and add it to `requestFileDeleteMap` if a pre-existing list cannot be found. However this method will not remove that list if there are no more items in the list after removal, which may lead to a memory leak. If `fileToDelete.isEmpty()` after an item is removed, shouldn't we also remove the associated entry from `requestFileDeleteMap`?

Also it seems strange to create a new list and add it to `requestFileDeleteMap` if it doesn't exist in this case. Would it make sense to just release the `request` and skip the add/remove from the `requestFileDeleteMap ` map?

Reply:
👍 Fixed